pkg/controller/deployment/deployment_controller.go

```diff
@@ -85,6 +85,8 @@ type DeploymentController struct {
 
 	// Deployments that need to be synced
 	queue workqueue.RateLimitingInterface
+	// Deployments that need to be checked for progress.
+	progressQueue workqueue.RateLimitingInterface
```

Comment:
Why have a secondary queue instead of simply adding using the `AddAfter` on the normal queue?